lib/oli/delivery/sections.ex

```diff
@@ -771,31 +771,94 @@ defmodule Oli.Delivery.Sections do
     create_section_resources(section, publication)
   end
 
+  @doc """
+  Retrieves all the publications an user can see:
+    - Associated to a community assigned to their institution
+    - Associated to a community they are assigned as a user
+    - If they have a linked author account
+      - They are an author of the publication
+      - An author has made publication visible to their institution
+      - Another author has shared publication with them
+    - Global -> will see them when they are not associated to any community, or one of the associated communities allows it
+
+  ## Examples
+
+      iex> retrieve_visible_publications(%User{}, %Institution{})
+      [%Publication{project: %Project{}}, ...]
+
+      iex> retrieve_visible_publications(%User{}, %Institution{})
+      []
+  """
+  def retrieve_visible_publications(user, institution) do
```

Comment:
I'd make the same comment here (about dependency cycles), but it looks like there was already a cycle in the code between `Oli.Delivery.Sections` and `Oli.Publishing`.  The intent is that Publishing depends on Sections context.